styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/ExecutionDescriptionHandler.java

```diff
@@ -61,12 +61,19 @@ public void transitionInto(RunState state) {
     switch (state.state()) {
       case PREPARE:
         try {
-          final Event createdEvent =
-              Event.submit(state.workflowInstance(), getExecDescription(workflowInstance));
-          try {
-            stateManager.receive(createdEvent);
-          } catch (StateManager.IsClosed isClosed) {
-            LOG.warn("Could not send 'created' event", isClosed);
+          final Optional<Workflow> workflow = storage.workflow(workflowInstance.workflowId());
+          if (workflow.isPresent()) {
```

Comment:
It is `ResourceNotFoundException ` misleading. We shouldn't throw this exception for missing a field of a resource. I will fix that.